src/status_im2/subs/wallet/wallet_test.cljs

```diff
@@ -176,3 +214,32 @@
          :removed                  false
          :balance                  3250}
         (rf/sub [sub-name])))))
+
+(h/deftest-sub :wallet/network-preference-details
+  [sub-name]
+  (testing "returns current viewing account address"
+    (swap! rf-db/app-db
+      #(-> %
+           (assoc-in [:wallet :accounts] accounts)
+           (assoc-in [:wallet :current-viewing-account-address] "0x1")
+           (assoc :wallet/tokens tokens)
+           (assoc :wallet/networks network-data)))
+    (is
+     (= [{:short-name       "eth"
+          :network-name     :ethereum
+          :chain-id         1
+          :related-chain-id nil
+          :layer            1}
+         {:short-name       "arb1"
+          :network-name     :arbitrum
+          :chain-id         42161
+          :related-chain-id nil
+          :layer            2}
+         {:short-name       "opt"
+          :network-name     :optimism
+          :chain-id         10
+          :related-chain-id nil
+          :layer            2}]
+        (->> (rf/sub [sub-name])
+             ;; Removed `#js source` property for correct compare
```

Comment:
why do we have `source`? what does it represents?
isn't it possible to add it to the mock db in this test?

Reply:
Here example value in test:
```:source #js {:uri "../resources/images/networks/Ethereum.png"}```

As I understand - comparison of js objects occurs not by value, but by reference, so they are not equal.

```
(=
   {:source #js {:uri "../resources/images/networks/Ethereum.png"}}
   {:source #js {:uri "../resources/images/networks/Ethereum.png"}})
  ;; => false
 
  (=
   {:source {:uri "../resources/images/networks/Ethereum.png"}}
   {:source {:uri "../resources/images/networks/Ethereum.png"}})
  ;; => true
```
Thats why I decided to omit this property in test